Fix for the DetectionPlot test failure

rita-gwen opened the pull request.

brendanx67 commented on pwiz_tools/Skyline/TestFunctional/DetectionsPlotTest.cs:
```diff
@@ -78,8 +78,9 @@ protected override void DoTest()
+                propDialog.OkDialog();
             });
-            OkDialog(propDialog, propDialog.OkDialog);
+            WaitForClosedForm(propDialog);
```
This is functionally equivalent to the code it replaces.

rita-gwen replied:
Saves one Invoke() call. My initial theory was that two invoke calls in quick succession might occasionally be executed out of order.  It proved to be incorrect, but since it's slightly more efficient than the original code I decided to keep it.

brendanx67 commented on pwiz_tools/Skyline/Controls/Graphs/DetectionsGraphController.cs:
```diff
@@ -153,7 +153,7 @@ public class Settings
             {
                 get => Properties.Settings.Default.DetectionsQValueCutoff;
-                set => Properties.Settings.Default.DetectionsQValueCutoff = value;
+                set => Properties.Settings.Default.DetectionsQValueCutoff = value; 
```
Probably not worth making this line-ending change, since it is the only change to this file.

brendanx67 commented on pwiz_tools/Skyline/TestFunctional/DetectionsPlotTest.cs:
```diff
@@ -91,29 +92,31 @@ protected override void DoTest()
+                propDialog.OkDialog();
             });
-            OkDialog(propDialog, propDialog.OkDialog);
+            WaitForClosedForm(propDialog);
```
Again. This doesn't actually change anything.

brendanx67 commented on pwiz_tools/Skyline/TestFunctional/DetectionsPlotTest.cs:
```diff
@@ -174,29 +179,29 @@ protected override void DoTest()
             DetectionPlotData data = null;
-            WaitForConditionUI(() => (data = pane.CurrentData) != null 
-                                           && data.QValueCutoff == qValue
+            WaitForConditionUI(() => pane.CurrentData != null 
```
Lots of unnecessary white space changes above. Is this the critical change? Using CurrentData everywhere instead of assigning it in the first logical expression and then using the assigned value?

brendanx67 commented on pwiz_tools/Skyline/Controls/Graphs/DetectionToolbarProperties.cs:
```diff
@@ -129,10 +129,10 @@ private void cmbTargetType_SelectedIndexChanged(object sender, EventArgs e)
         {
             if (qValue == .01f)
-                rbQValue01.Select();
+                rbQValue01.Checked = true;
```
Is this the critical change? It does seem wrong to use Select() instead of Checked = true. Not sure why that would pass sometimes and not others, though.

rita-gwen replied:
The only way I was able to reproduce the timeout failure was to focus on another window while the test was running and that traced down to using Select() method. I cannot be 100% sure, but I ran this test in a loop for an hour without any exceptions.

brendanx67 reviewed (changes requested):
Not super clear what the fix is. Looks like a lot of white space changes and other changes that I would expect not to make any difference.

rita-gwen commented:
Sorry about the whitespace changes. I added and removed a lot of tracing statements while debugging and Visual Studio is very trigger-happy to re-format the surrounding code.

rita-gwen commented on pwiz_tools/Skyline/TestFunctional/DetectionsPlotTest.cs:
```diff
@@ -174,29 +179,29 @@ protected override void DoTest()
                 () => $"Retrieving data for qValue {qValue}, refIndex {refIndex} took too long.");
             WaitForGraphs();
-            Assert.IsTrue(data.IsValid);
+            WaitForCondition(() => pane.CurrentData.IsValid);
```
This is the fix for the assertion failure that Nick tried to fix. The problem was that WaitForGraphs() doesn't guarantee that the data computation is complete. I've seen this in the RT regression tests as well.
